modules/cnn_3dobj/include/opencv2/cnn_3dobj.hpp

```diff
@@ -0,0 +1,252 @@
+        /** @brief Make all view points having the some distance from the focal point used by the camera view.
+         */
+        CV_WRAP void norm(float v[]);
+        /** @brief Add new view point between 2 point of the previous view point.
+         */
+        CV_WRAP void add(float v[]);
+        /** @brief Generating new view points from all triangles.
+         */
+        CV_WRAP void subdivide(float v1[], float v2[], float v3[], int depth);
+
+        public:
+        /** @brief Camera position on the sphere after duplex position elimination.
+         */
+        std::vector<cv::Point3d> CameraPos;
+        /** @brief Generating a sphere by mean of a iteration based points selection process.
+        @param radius_in Another radius used for adjusting the view distance.
+        @param depth_in Number of interation depth for increasing the points on sphere.
+         */
+        icoSphere(float radius_in, int depth_in);
+        /** @brief Get the center of points on surface in .ply model.
+        @param cloud Point cloud used for computing the center point.
+         */
+        CV_WRAP cv::Point3d getCenter(cv::Mat cloud);
+        /** @brief Get the proper camera radius from the view point to the center of model.
+        @param cloud Point cloud used for computing the center point.
+        @param center center point of the point cloud.
+         */
+        CV_WRAP float getRadius(cv::Mat cloud, cv::Point3d center);
+        /** @brief Suit the position of bytes in 4 byte data structure for particular system.
+         */
+        CV_WRAP static uint32_t swapEndian(uint32_t val);
+        /** @brief Create header in binary files collecting the image data and label.
+        @param num_item Number of items.
+        @param rows Rows of a single sample image.
+        @param cols Columns of a single sample image.
+        @param headerPath Path where the header will be stored.
+         */
+        CV_WRAP static void createHeader(int num_item, int rows, int cols, const char* headerPath);
+        /** @brief Write binary files used for training in other open source project including Caffe.
+        @param filenameImg Path which including a set of images.
+        @param binaryPath Path which will output a binary file.
+        @param headerPath Path which header belongs to.
+        @param num_item Number of samples.
+        @param label_class Class label of the sample.
+        @param x Pose label of X.
+        @param y Pose label of Y.
+        @param z Pose label of Z.
+        @param isrgb Option for choice of using RGB images or not.
+         */
+        CV_WRAP static void writeBinaryfile(string filenameImg, const char* binaryPath, const char* headerPath, int num_item, int label_class, int x, int y, int z, int isrgb);
+    };
+
+/** @brief Caffe based 3D images descriptor.
+ The class extract feature which is discriminant for class and pose from given samples @cite wohlhart15.
```

Comment:
Replace:
A class to extract features from an image. The so obtained descriptors can be used for classification and pose estimation goals @cite wohlhart15.